plugins/auto-sizes/includes/improve-calculate-sizes.php

```diff
@@ -135,50 +130,176 @@ function auto_sizes_filter_image_tag( $content, array $parsed_block, WP_Block $b
 /**
  * Modifies the sizes attribute of an image based on layout context.
  *
- * @param int    $id           The image id.
- * @param string $size         The image size data.
- * @param string $align        The image alignment.
- * @param string $resize_width Resize image width.
- * @return string The sizes attribute value.
+ * @since n.e.x.t
+ *
+ * @param int          $id            The image id.
+ * @param string|int[] $size          The image size data.
+ * @param string       $align         The image alignment.
+ * @param string       $resize_width  Resize image width.
+ * @param string       $max_alignment The maximum usable layout alignment.
+ * @return string|false An improved sizes attribute or false if a better size cannot be calculated.
  */
-function auto_sizes_calculate_better_sizes( int $id, string $size, string $align, string $resize_width ): string {
-	$sizes = '';
-	$image = wp_get_attachment_image_src( $id, $size );
+function auto_sizes_calculate_better_sizes( int $id, $size, string $align, string $resize_width, string $max_alignment ) {
```

Comment:
```suggestion
function auto_sizes_calculate_better_sizes( int $id, $size, string $align, int $resize_width, string $max_alignment ) {
```

The phpdoc would need to be updated as well.